runtime/Makefile

```diff
@@ -46,22 +58,14 @@ examples: | runtime
 	$(MAKE) -C examples all
 
 .PHONY: test
-test: | runtime
+test: | configure
 	@echo "test the Catalyst runtime test suite"
 	cmake --build $(RT_BUILD_DIR) --target runner_tests -j$(if $(nprocs:-=),$(nprocs),$$(nproc))
 	$(RT_BUILD_DIR)/tests/runner_tests
 
 .PHONY: coverage
-coverage: | qir
+coverage: configure | qir
 	@echo "check C++ code coverage"
-	cmake -G Ninja -B $(RT_BUILD_DIR) . \
-		-DCMAKE_BUILD_TYPE=Debug \
-		-DENABLE_LIGHTNING_KOKKOS=$(if $(kokkos:-=),$(kokkos),OFF) \
-		-DCMAKE_C_COMPILER=$(C_COMPILER) \
-		-DCMAKE_CXX_COMPILER=$(CPP_COMPILER) \
-		-DQIR_STDLIB_LIB=$(QIR_STDLIB_DIR) \
-		-DQIR_STDLIB_INCLUDES=$(QIR_STDLIB_INCLUDES_DIR) \
-		-DENABLE_CODE_COVERAGE=ON
 
 	cmake --build $(RT_BUILD_DIR) --target runner_tests -j$(if $(nprocs:-=),$(nprocs),$$(nproc))
 	$(RT_BUILD_DIR)/tests/runner_tests
```

Comment:
It looks like this new target doesn't do anything, or does it still work?

Reply:
Line 70 builds the runtime with the specified configuration and with the target `runner_tests`.

Reply:
Right but it does the same thing as the `test` target, rather than ensuring we are running the coverage.

Reply:
There is more beyond line 67 and 71. There are calls to `lcov` which will generate the coverage file. So, slightly more work done besides the `test` target.

Reply:
That is true, my main concern was that this target no longer sets the `-DENABLE_CODE_COVERAGE=ON` flag.

Reply:
It should. As stated in another [comment](https://github.com/PennyLaneAI/catalyst/pull/41#discussion_r1129983560):

```
coverage: CODE_COVERAGE=ON
```

[This is how to set variables for specific targets.](https://www.gnu.org/software/make/manual/html_node/Target_002dspecific.html)

Reply:
Oh awesome, that's great! You ignore this comment then :)